Formula/bettercap.rb

```diff
@@ -1,39 +1,31 @@
 class Bettercap < Formula
-  desc "Complete, modular, portable and easily extensible MITM framework"
+  desc "It's is the swiss army knife for network attacks and monitoring"
   homepage "https://www.bettercap.org/"
-  url "https://github.com/evilsocket/bettercap/archive/v1.6.2.tar.gz"
-  sha256 "1b364d7e31be5fa7b5f93eefe76763ad7bd4ac0b7b6bb4af05483157580a9cb9"
-  revision 3
-
-  bottle do
-    cellar :any
-    sha256 "f37cb67f4deaf5433c117339ea68350c01f88516001484443872cb4d4be830b5" => :high_sierra
-    sha256 "5f6ed37beb8ad8525eb5fc226d17a8fe5893ecb29a0733347cfb5a7f8a2ade7f" => :sierra
-    sha256 "c15257bc4fcc3d23bdb6e1204757230f6ca843b97507ef4ceefa824e8ec4838a" => :el_capitan
-  end
+  url "https://github.com/bettercap/bettercap/archive/v2.1.tar.gz"
+  sha256 "fb3958e9f8b677ad2293fd5542f7c250613abec285ad490a519017dce9ffb8fc"
+  head "https://github.com/bettercap/bettercap"
 
   depends_on "openssl"
-  depends_on "ruby" if MacOS.version <= :sierra
+  depends_on "libpcap"
+  depends_on "go" => :build
 
   def install
-    ENV["GEM_HOME"] = libexec
-    ENV["BUNDLE_PATH"] = libexec
-    ENV.prepend "CPPFLAGS", "-I#{Formula["openssl"].opt_include}"
-    system "gem", "install", "bundler"
-    system libexec/"bin/bundle", "install"
-    system "gem", "build", "bettercap.gemspec"
-    system "gem", "install", "bettercap-#{version}.gem"
-    bin.install libexec/"bin/bettercap"
-    bin.env_script_all_files(libexec/"bin", :GEM_HOME => ENV["GEM_HOME"])
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/bettercap/bettercap").install buildpath.children
+    cd "src/github.com/bettercap/bettercap" do
+      system "make", "deps"
```

Comment:
This is running `go get`, which we don't allow.

Please create an issue upstream requesting go vendoring.

Reply:
why? installing dependencies it's as easy as `go get ./...`, why are you forcing people to use an additional tool in their build pipeline in order to be integrated here?

Reply:
@evilsocket The build needs to reproducible and secure, neither of which is the case if you use `go get`. In order to make the build reproducible without `go get` and without upstream vendoring being used, we'd need to add the following to this formula:
```
  go_resource "github.com/adrianmo/go-nmea" do
    url "https://github.com/adrianmo/go-nmea.git",
        :revision => "c227bc33b4c6213c7f2ae4e2f69d0c9d37bea7dc"
  end

  go_resource "github.com/bettercap/readline" do
    url "https://github.com/bettercap/readline.git",
        :revision => "9cec905dd29109b64e6752507fba73474c2efd46"
  end

  go_resource "github.com/dustin/go-humanize" do
    url "https://github.com/dustin/go-humanize.git",
        :revision => "bb3d318650d48840a39aa21a027c6630e198e626"
  end

  go_resource "github.com/elazarl/goproxy" do
    url "https://github.com/elazarl/goproxy.git",
        :revision => "a96fa3a318260eab29abaf32f7128c9eb07fb073"
  end

  go_resource "github.com/google/gopacket" do
    url "https://github.com/google/gopacket.git",
        :revision => "b40365fca6ccb41d3892c4caa0f6e7545ee2fdcf"
  end

  go_resource "github.com/gorilla/websocket" do
    url "https://github.com/gorilla/websocket.git",
        :revision => "eb925808374e5ca90c83401a40d711dc08c0c0f6"
  end

  go_resource "github.com/inconshreveable/go-vhost" do
    url "https://github.com/inconshreveable/go-vhost.git",
        :revision => "06d84117953b22058c096b49a429ebd4f3d3d97b"
  end

  go_resource "github.com/malfunkt/iprange" do
    url "https://github.com/malfunkt/iprange.git",
        :revision => "3a31f5ed42d2d8a1fc46f1be91fd693bdef2dd52"
  end

  go_resource "github.com/mattn/go-isatty" do
    url "https://github.com/mattn/go-isatty.git",
        :revision => "6ca4dbf54d38eea1a992b3c722a76a5d1c4cb25c"
  end

  go_resource "github.com/mdlayher/dhcp6" do
    url "https://github.com/mdlayher/dhcp6.git",
        :revision => "e26af0688e455a82b14ebdbecf43f87ead3c4624"
  end

  go_resource "github.com/pkg/errors" do
    url "https://github.com/pkg/errors.git",
        :revision => "30136e27e2ac8d167177e8a583aa4c3fea5be833"
  end

  go_resource "github.com/robertkrimen/otto" do
    url "https://github.com/robertkrimen/otto.git",
        :revision => "6c383dd335ef8dcccef05e651ce1eccfe4d0f011"
  end

  go_resource "github.com/tarm/serial" do
    url "https://github.com/tarm/serial.git",
        :revision => "eaafced92e9619f03c72527efeab21e326f3bc36"
  end

  go_resource "gopkg.in/sourcemap.v1" do
    url "https://gopkg.in/sourcemap.v1.git",
        :revision => "6e83acea0053641eff084973fee085f0c193c61a"
  end
```
Then those resource blocks need to be updated every release. It's much more sustainable for the package to have those Git commit hashes managed by upstream rather than inferred by us just-in-time and stored in the formula file.

No additional tooling is actually required since `go` natively supports the presence of a `vendor` directory. However, [best practice is to use some tooling](https://peter.bourgon.org/go-best-practices-2016/?utm_source=hackernewsletter&utm_medium=email&utm_term=code#dependency-management) to manage it whether it's actually checked into source control or not. See https://research.swtch.com/vgo-intro which recommends `dep` unless a project is already able to use `vgo`. For our purposes, `dep`, `glide`, `govendor`, `godep`, etc. or a manually managed `vendor/` directory are all fine. Some projects use a separate release tarball for this. See https://github.com/Homebrew/homebrew-core/blob/master/Formula/lxc.rb#L4

Reply:
"Use any deps manager" is different than "use dep" :) Fair enough, will see which one is the easier to integrate.

Reply:
We're using glide now.